### PR TITLE
Update Release Notes for oneDPL 2022.6.0 release

### PR DESCRIPTION
**Pull request by ValentinaKats**
Added oneDPL 2022.6.0 Release Notes

### CODE REVIEW

File: documentation/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,98 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+News
+------------
+- `oneAPI DPC++ Library Manual Migration Guide`_ to simplify the migration of Thrust* and CUB* APIs from CUDA*. 
+- ``radix_sort`` and ``radix_sort_by_key`` algorithms residing in the ``oneapi::dpl::experimental::kt::esimd`` namespace
+  were moved into ``oneapi::dpl::experimental::kt::gpu::esimd`` namespace. 
+  The former namespace is now deprecated and will be removed in the future release.
+
+New Features
+------------
+- oneDPL code is now implemented on tag dispatching approach.
```
Comment (MikeDvorskiy) [inline documentation/release_notes.rst:22]: "tag dispatching" is a kind of oneDPL implementation details. I think we should not mention that in the release notes.

File: documentation/release_notes.rst [hunk 2]
```diff
@@ -8,6 +8,98 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+  The former namespace is now deprecated and will be removed in the future release.
+
+New Features
+------------
+- oneDPL code is now implemented on tag dispatching approach. 
+  ``for_loop``, ``for_loop_strided``, ``for_loop_n``,  ``for_loop_n_strided`` are implemented for the host policies only. 
+- Added experimental ``inclusive_scan`` algorithm residing in
+  the ``oneapi::dpl::experimental::kt::gpu`` namespace. This algorithm is part of
+  the family of kernel templates that allow configuring a variety of parameters
+  including the number of elements to process by a work item, and the size of a workgroup.
```
[the remark this answers is not in the document]
Comment (adamfidel) [reply documentation/release_notes.rst:27]: I agree with this suggestion. I don't think a footnote would be necessary here.

File: documentation/release_notes.rst [hunk 3]
```diff
@@ -8,6 +8,96 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+- `oneAPI DPC++ Library Manual Migration Guide`_ to simplify the migration of Thrust* and CUB* APIs from CUDA*. 
+- ``radix_sort`` and ``radix_sort_by_key`` kernel templates were moved into
+  ``oneapi::dpl::experimental::kt::gpu::esimd`` namespace. The former ``oneapi::dpl::experimental::kt::esimd``
+  namespace is deprecated and will be removed in a future release.
+
+New Features
+------------
+- oneDPL code is now implemented on tag dispatching approach. 
+- The ``for_loop``, ``for_loop_strided``, ``for_loop_n``,  ``for_loop_n_strided`` algorithms
+  in `namespace oneapi::dpl::experimental` are enforced to fail with device execution policies.
```
Comment (akukanov) [inline documentation/release_notes.rst:24]: If we keep this note, let's then move it to the "News" section above.

File: documentation/release_notes.rst [hunk 4]
```diff
@@ -8,6 +8,98 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+- Fixed compilation errors when using ``reduce``, ``min_element``, ``max_element``,
+  ``minmax_element``, ``is_partitioned``, and ``lexicographical_compare`` with DPCPP compilers 2023.0 and below.
+- Fixed possible data races in the following algorithms used with DPC++ execution policies: ``remove_if ``, ``unique ``,
+  ``inplace_merge ``, `` stable_partition``, `` partial_sort_copy``, `` rotate``.
+- Fixed handling of ``std::vector`` iterators allocated with USM allocators as input to oneDPL algorithms using
+  a device policy for standard library implementations which include allocator information in the iterator type.
+- Fixed an issue with ``transform_iterator`` where checking ``std::is_default_constructible`` of a `transform_iterator``
+  with a functor which is not default constructible would result in a build error or an incorrect result for some compilers
+  and implementations of the standard library. Added copy assignment of the functor within a ``transform_iterator``
+  when the functor is copy assignable.
```
[the remark this answers is not in the document]
Comment (ValentinaKats) [reply documentation/release_notes.rst]: @danhoeflinger, @akukanov I applied Alexey's suggestion. Please let me know if it needs further modifications.

[the remark this answers is not in the document]
Comment (danhoeflinger) [reply documentation/release_notes.rst]: I think Alexey's suggestion is good without modification, thanks.

File: documentation/release_notes.rst [hunk 5]
```diff
@@ -8,6 +8,98 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+- Fixed compilation errors when using ``oneapi::dpl::zip_iterator`` with the oneTBB backend and C++20.
+
+Known Issues and Limitations
+----------------------------
+New in This Release
+^^^^^^^^^^^^^^^^^^^
+- ``histogram`` algorithm requires the output value type to be an integral type no larger than 4 bytes when used with an FPGA policy.
+- ``reduce_by_segment``, when used with C++ standard policies, imposes limitations on the value type.
+  * Firstly, it must satisfy the ``DefaultConstructible`` requirements.
+  * Secondly, a default-constructed instance of that type should act as the identity element for the binary reduction function.
```
[the remark this answers is not in the document]
Comment (akukanov) [reply documentation/release_notes.rst:57]:
```suggestion
```

**Review by akukanov — APPROVED** [review]

**Pull request merged**